mmv1/products/iamworkforcepool/WorkforcePoolProvider.yaml

```diff
@@ -252,6 +252,30 @@ properties:
           The client ID. Must match the audience claim of the JWT issued by the
           identity provider.
         required: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'clientSecret'
+        description: |
+          The optional client secret. Required to enable Authorization Code flow for web sign-in.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'value'
+            exactly_one_of:
+              - oidc.0.client_secret.0.value
+            description: |
+              The value of the client secret.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'plainText'
+                description: |
+                  The plain text of the client secret value.
+                sensitive: true
+                required: true
+                custom_flatten: templates/terraform/custom_flatten/iam_workforce_pool_provider_oidc_client_secret_value.go.erb
+              - !ruby/object:Api::Type::String
+                name: 'thumbprint'
```

Comment:
Can we make use of the server-computed `thumbprint` of the client secret to trigger a diff if the client secret is updated externally outside of Terraform? That is, trigger a diff if any of these is true:

1. Within Terraform, user changes client secret plain text. (Note that the client secret plainText is not returned from the server, so we used `custom_flatten` above)
2. Terraform sees server API response's `thumbprint` field is different from the one stored in the state.

Reply:
That will require a customize diff function to be written by hand to check the local value of thumbprint against the API's new value and then trigger a change on the plain value

Reply:
Thanks, I tried `custom_diff` but it cannot get changes due to computed fields (in https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/customizing-differences it says that I believe). However I think I managed to get the drift correction to work using `custom_flatten`. PTAL.